src/reader/mod.rs

```diff
@@ -346,18 +346,28 @@ macro_rules! read_until_close {
                 }
             },
             // `</` - closing tag
+            // #776: We parse using ElementParser which allows us to have attributes
+            // in close tags. While such tags are not allowed by the specification,
+            // we anyway allow to parse them because:
+            // - we do not check constrains during parsing. This is performed by the
+            //   optional validate step which user should call manually
+            // - if we just look for `>` we will parse `</tag attr=">" >` as end tag
+            //   `</tag attr=">` and text `" >` which probably no one existing parser
+            //   do. This is still invalid tag but it also has no practical meaning
+            //   to parse it in a such way. End tag `</tag attr=">" >`, however, can
+            //   be produced by some real-world parsers, for example, by the parser
+            //   of the Macromedia Flash player.
```

Comment:
For what it's worth, other "non-validating" parsers like expat still blow up on this

>ExpatError: not well-formed (invalid token): line 1, column 11

Lxml and libxml2 also do

>XMLSyntaxError: expected '>', line 1, column 12

>parserError: xmlParseDoc() failed

I'm on-board with making this possible to support, but our default behavior should probably be throwing an `IllFormedError` just like we do for mismatched tags (and like other parsers do).

>ExpatError: mismatched tag: line 1, column 7

Reply:
I want to evolve library in this direction:
- rename `Reader` to `RawReader` (and `Event` to `RawEvent`)
- make `RawReader` as liberal as possible. It should not make any validation checks
- add a `validate` method to `RawEvent` which will perform all well-formedless and validity checks (I already grab list of all checks from the [specification](https://www.w3.org/TR/xml11), 35 VC + 16 WFC references, actually most of them about DTD). All validation must be run manually by calling `RawEvent::validate`. Only one check cannot be performed in this way: check that [start tag name matches end tag name](https://www.w3.org/TR/xml11/#GIMatch)
- move existing checks (for example, [`IllFormedError::DoubleHyphenInComment`](https://github.com/tafia/quick-xml/blob/22b3e458d193e6937882ba6b643ec8c31a671de5/src/errors.rs#L119C26-L119C27)) to the new `validate` method
- introduce a new `Reader` which will perform validation checks when flag in config is set (and process many other things, like dealing with encodings and expanding entity references as shown in the `custom_entities` example in #766). Add necessary config flags

I haven't thought through all the details yet and don't want to focus on them yet. There will be a separate PR where all this will be considered. It should be pretty soon (this year).

So, given that we are not currently performing most validity checks and I am planning a generic consistent solution for them, do you agree that it is not worth introducing them now?

Reply:
> Also line 358 should be rewritten - XML isn't "produced" by a parser. You can just say that such documents exist in the wild and are tolerated by some parsers such as the one used by Adobe Flash.

Any help are appreciated :smiley: . I feel that my English is no good to writing simple-to-read explanations.

Reply:
That all sounds good to me, but we probably shouldn't document it as a "new feature" given that we do want this to eventually fail for users of the `Reader` API.